packages/web/src/common/store/upload/sagas.js

```diff
@@ -1015,18 +1012,25 @@ function* uploadSingleTrack(track) {
           throw new Error(`Could not confirm upload single track ${trackId}`)
         }
 
-        // getting track immediately after acdc confirms
-        // can 404 because it's not indexed yet
-        yield delay(1500)
-
-        return yield apiClient.getTrack({
-          id: trackId,
-          currentUserId: userId,
-          unlistedArgs: {
-            urlTitle: formatUrlName(track.metadata.title),
-            handle
+        let retries = 30
+        while (retries !== 0) {
+          const maybeTrackRes = yield apiClient.getTrack({
+            id: trackId,
+            currentUserId: userId,
+            unlistedArgs: {
+              urlTitle: formatUrlName(track.metadata.title),
+              handle
+            }
+          })
+          if (maybeTrackRes !== undefined && maybeTrackRes !== null) {
+            return maybeTrackRes
           }
-        })
+          // lil bit of delay for retries
+          yield delay(1500)
```

Comment:
Should we use a variable delay to disperse groups of failed calls?

Reply:
like add a back off of some kind?

Reply:
Yeah... I'm not sure if it's necessary though. Concurrent upload volume shouldn't be high enough to stress anything on a 1.5s interval.

Reply:
I wouldn't think so. This is also a `GET` query, not an actual write operation.

Reply:
ya let's leave as is. Ty for entertaining my thinking out loud lol